Content.Client/Xenoarchaeology/Ui/AnalysisConsoleMenu.xaml.cs

```diff
@@ -15,169 +24,235 @@ namespace Content.Client.Xenoarchaeology.Ui;
+
+        if (_playerManager.LocalSession?.AttachedEntity is { } attachedEntity)
+        {
+            var volume = _nodeExtractionValues.Count == 0 ? 1f : -10f;
+            _audio.PlayGlobal(_owner.Comp.ScanFinishedSound, attachedEntity, AudioParams.Default.WithVolume(volume));
+        }
+
+        if (_nodeExtractionValues.Count == 0)
+            OnExtractButtonPressed?.Invoke();
     }
 
-    public void SetButtonsDisabled(AnalysisConsoleUpdateState state)
+    public void Update(Entity<AnalysisConsoleComponent> ent)
     {
-        ScanButton.Disabled = !state.CanScan;
-        PrintButton.Disabled = !state.CanPrint;
-        if (state.IsTraversalDown)
-            DownBiasButton.Pressed = true;
-        else
-            UpBiasButton.Pressed = true;
+        _artifactAnalyzer.TryGetArtifactFromConsole(ent, out var arti);
+        ArtifactView.SetEntity(arti);
+        GraphControl.SetArtifact(arti);
+
+        ExtractButton.Disabled = arti == null;
+
+        if (arti == null)
+            NoneSelectedLabel.Visible = false;
 
-        ExtractButton.Disabled = false;
-        if (!state.ServerConnected)
+        NoArtiLabel.Visible = true;
+        if (!_artifactAnalyzer.TryGetAnalyzer(ent, out _))
         {
-            ExtractButton.Disabled = true;
-            ExtractButton.ToolTip = Loc.GetString("analysis-console-no-server-connected");
+            NoArtiLabel.Text = Loc.GetString("analysis-console-info-no-scanner");
         }
-        else if (!state.CanScan)
+        else if (arti == null)
         {
-            ExtractButton.Disabled = true;
-
-            // CanScan can be false if either there's no analyzer connected or if there's
-            // no entity on the scanner. The `Information` text will always tell the user
-            // of the former case, but in the latter, it'll only show a message if a scan
-            // has never been performed, so add a tooltip to indicate that the artifact
-            // is gone.
-            if (state.AnalyzerConnected)
-            {
-                ExtractButton.ToolTip = Loc.GetString("analysis-console-no-artifact-placed");
-            }
-            else
-            {
-                ExtractButton.ToolTip = null;
-            }
+            NoArtiLabel.Text = Loc.GetString("analysis-console-info-no-artifact");
         }
-        else if (state.PointAmount <= 0)
+        else
         {
-            ExtractButton.Disabled = true;
-            ExtractButton.ToolTip = Loc.GetString("analysis-console-no-points-to-extract");
+            NoArtiLabel.Visible = false;
         }
 
-        if (ExtractButton.Disabled)
-        {
-            ExtractButton.RemoveStyleClass("ButtonColorGreen");
-        }
-        else
+        if (_currentNode == null
+            || arti == null
+            || !_xenoArtifact.TryGetIndex((arti.Value, arti.Value), _currentNode.Value, out _))
         {
```

Comment:
Because there is some code that Emo made and there are some things that i corrected :D And you can meet it around all PR sadly. But here it was with braces also because if block is multiline. I'll  remove braces here to honor your comment tho :3 i checked whole file - other places have braces with ifs only when justified 👍